### PR TITLE
🏃cmd-clusterctl-client-cluster/tests: standardize gomega imports

### PR DESCRIPTION
**Pull request by cpanato**
**What this PR does / why we need it**:
standardize gomega imports for `cmd/clusterctl/client/cluster/**` tests
will open other PR for the other folders to make easier the review

/cc @vincepri @detiber 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related to https://github.com/kubernetes-sigs/cluster-api/issues/2433

### CODE REVIEW

File: cmd/clusterctl/client/cluster/objectgraph_test.go [hunk 1]
```diff
@@ -98,23 +100,15 @@ type wantGraph struct {
...
-		if gotNode.virtual != wantNode.virtual {
-			t.Errorf("node with uid = %s, got virtual = %t, want %t", uid, gotNode.virtual, wantNode.virtual)
-		}
-
-		if len(gotNode.owners) != len(wantNode.owners) {
-			t.Fatalf("node with uid = %s, got owners = %d, want %d", uid, len(gotNode.owners), len(wantNode.owners))
-		}
+		g.Expect(ok).To(BeTrue())
+		g.Expect(gotNode.virtual).To(Equal(wantNode.virtual))
+		g.Expect(len(gotNode.owners)).To(Equal(len(wantNode.owners)))
```
Comment (detiber) [inline cmd/clusterctl/client/cluster/objectgraph_test.go]:
```suggestion
		g.Expect(gotNode.owners).To(HaveLen(len(wantNode.owners)))
```

File: cmd/clusterctl/client/cluster/objectgraph_test.go [hunk 2]
```diff
@@ -1127,15 +1107,15 @@ func Test_objectGraph_setSoftOwnership(t *testing.T) {
 				sort.Strings(wantObjects)
 				sort.Strings(gotObjects)
 
-				if !reflect.DeepEqual(gotObjects, wantObjects) {
-					t.Fatalf("secret %s, got = %s, expected = %s", secret.identity.UID, gotObjects, wantObjects)
-				}
+				g.Expect(gotObjects).To(Equal(wantObjects))
```
Comment (detiber) [inline cmd/clusterctl/client/cluster/objectgraph_test.go]:
```suggestion
				g.Expect(gotObjects).To(ConsistOf(wantObjects))
```
This also removes the need to call `sort.Strings()` above for `wantObjects` and `gotObjects`

File: cmd/clusterctl/client/cluster/objectgraph_test.go [hunk 3]
```diff
@@ -1303,9 +1277,7 @@ func Test_objectGraph_setClusterTenants(t *testing.T) {
 				sort.Strings(wantTenants)
 				sort.Strings(gotTenants)
 
-				if !reflect.DeepEqual(gotTenants, wantTenants) {
-					t.Fatalf("cluster %s, got = %s, expected = %s", cluster.identity.UID, gotTenants, wantTenants)
-				}
+				g.Expect(gotTenants).To(Equal(wantTenants))
```
Comment (detiber) [inline cmd/clusterctl/client/cluster/objectgraph_test.go]:
```suggestion
				g.Expect(gotTenants).To(ConsistOf(wantTenants))
```
Also can remove the `sort.Strings()` above.

File: cmd/clusterctl/client/cluster/template_test.go [hunk 4]
```diff
@@ -209,31 +206,28 @@ func Test_templateClient_getGitHubFileContent(t *testing.T) {
...
-	}
+	g.Expect(err).NotTo(HaveOccurred())
 	defer os.RemoveAll(tmpDir)
 
 	path := filepath.Join(tmpDir, "cluster-template.yaml")
-	if err := ioutil.WriteFile(path, []byte(template), 0644); err != nil {
-		t.Fatalf("err: %s", err)
-	}
+	err = ioutil.WriteFile(path, []byte(template), 0644)
+	g.Expect(err).NotTo(HaveOccurred())
```
Comment (detiber) [inline cmd/clusterctl/client/cluster/template_test.go]:
```suggestion
	g.Expect(ioutil.WriteFile(path, []byte(template), 0644)).To(Succeed())
```

File: cmd/clusterctl/client/cluster/template_test.go [hunk 5]
```diff
@@ -307,9 +297,8 @@ func Test_templateClient_GetFromURL(t *testing.T) {
 	})
 
 	path := filepath.Join(tmpDir, "cluster-template.yaml")
-	if err := ioutil.WriteFile(path, []byte(template), 0644); err != nil {
-		t.Fatalf("err: %s", err)
-	}
+	err = ioutil.WriteFile(path, []byte(template), 0644)
+	g.Expect(err).NotTo(HaveOccurred())
```
Comment (detiber) [inline cmd/clusterctl/client/cluster/template_test.go]:
```suggestion
	g.Expect(ioutil.WriteFile(path, []byte(template), 0644)).To(Succeed())
	g.Expect(err).NotTo(HaveOccurred())
```

File: cmd/clusterctl/client/cluster/components_test.go [hunk 6]
```diff
@@ -253,17 +257,15 @@ func Test_providerComponents_Delete(t *testing.T) {
 				IncludeNamespace: tt.args.includeNamespace,
 				IncludeCRDs:      tt.args.includeCRD,
 			})
-			if (err != nil) != tt.wantErr {
-				t.Fatalf("error = %v, wantErr %v", err, tt.wantErr)
-			}
+			g.Expect(err != nil).To(Equal(tt.wantErr))
```
[the remark this answers is not in the document]
Comment (cpanato) [reply cmd/clusterctl/client/cluster/components_test.go]: forgot that :( my bad
done

File: cmd/clusterctl/client/cluster/installer_test.go [hunk 7]
```diff
@@ -317,17 +326,13 @@ func Test_shouldInstallSharedComponents(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := shouldInstallSharedComponents(tt.args.providerList, tt.args.provider)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
+			g.Expect(err != nil).To(Equal(tt.wantErr))
```
[the remark this answers is not in the document]
Comment (cpanato) [reply cmd/clusterctl/client/cluster/installer_test.go]: my bad as well :(
done

File: cmd/clusterctl/client/cluster/inventory_managementgroup_test.go [hunk 8]
```diff
@@ -147,17 +150,13 @@ func Test_inventoryClient_GetManagementGroups(t *testing.T) {
 				proxy: tt.fields.proxy,
 			}
 			got, err := p.GetManagementGroups()
-			if (err != nil) != tt.wantErr {
-				t.Errorf("error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
+			g.Expect(err != nil).To(Equal(tt.wantErr))
```
[the remark this answers is not in the document]
Comment (cpanato) [reply cmd/clusterctl/client/cluster/inventory_managementgroup_test.go]: done

File: cmd/clusterctl/client/cluster/inventory_test.go [hunk 9]
```diff
@@ -167,28 +171,28 @@ func Test_inventoryClient_Create(t *testing.T) {
...
+			g.Expect(err).NotTo(HaveOccurred())
+
 			for i := range got.Items {
 				tt.wantProviders[i].ResourceVersion = got.Items[i].ResourceVersion
 			}
 
-			if !reflect.DeepEqual(got.Items, tt.wantProviders) {
-				t.Errorf("got = %v, want %v", got.Items, tt.wantProviders)
-			}
+			g.Expect(got.Items).To(Equal(tt.wantProviders))
```
[the remark this answers is not in the document]
Comment (cpanato) [reply cmd/clusterctl/client/cluster/inventory_test.go]: thanks!

File: cmd/clusterctl/client/cluster/mover_test.go [hunk 10]
```diff
@@ -391,15 +388,14 @@ func Test_getMoveSequence(t *testing.T) {
 				sort.Strings(gotNodes)
 				sort.Strings(wantGroup)
 
-				if !reflect.DeepEqual(gotNodes, wantGroup) {
-					t.Errorf("group[%d], got = %s, expected = %s", i, gotNodes, wantGroup)
-				}
+				g.Expect(gotNodes).To(Equal(wantGroup))
```
[the remark this answers is not in the document]
Comment (cpanato) [reply cmd/clusterctl/client/cluster/mover_test.go]: done

**Comment by cpanato** [comment]
thanks @detiber for your review! PTAL

**Pull request merged**